frame/system/src/lib.rs

```diff
@@ -1442,6 +1442,14 @@ impl<T: Config> Pallet<T> {
 		Self::read_events_no_consensus().map(|e| *e).collect()
 	}
 
+	/// Get a single event at specified index.
+	///
+	/// Should only be called if you know what you are doing and outside of the runtime block
+	/// execution else it can have a large impact on the PoV size of a block.
+	pub fn event_no_consensus(index: usize) -> Option<T::RuntimeEvent> {
+		Events::<T>::get().get(index).map(|e| e.event.clone())
+	}
+
```

Comment:
I could just get a Vec of events and search in it. This function is for future proofing. IIRC there were plans to move from storing events in a vec to something more fine-grained. In case this will be implemented, it will be easier to fix all cases where accesing only a single event is required.

Reply:
I'll keep the function, but use the iterator internally.

Reply:
Done